### DISCUSSION TITLE
How to run Mockoon environment file on docker

### DISCUSSION THREAD

**Discussion by parulbhagat-soprabanking** [mockoon/mockoon#1056] [Help] [answered]
I have tried to mount local Mockoon environment file on docker container with mockoon-cli:latest image. It was building successfully but when it had started with in 3 seconds it stopped again and again. I have follow the steps as mentioned in the given URL

https://mockoon.com/tutorials/run-mock-api-anywhere-cli/

I have also tried with dockerfile
```
FROM node:18-alpine

RUN npm install -g @mockoon/cli@3.0.0
COPY mockoon-frontier-mssws-mocks.json ./mockoon-frontier-mssws-mocks.json

# Do not run as root.
RUN adduser --shell /bin/sh --disabled-password --gecos "" mockoon
RUN chown -R mockoon ./mockoon-frontier-mssws-mocks.json
USER mockoon

EXPOSE 3000

ENTRYPOINT ["mockoon-cli", "start", "--daemon-off", "--disable-log-to-file", "--data", "mockoon-frontier-mssws-mocks.json", "--container"]

# Usage: docker run -p <host_port>:<container_port> mockoon-test
```

but again facing same issue
I have followed the steps as mentioned in the given URL

https://mockoon.com/tutorials/run-mock-api-anywhere-cli/

**Comment by 255kb** [comment, marked as answer]
Do you get any error message in Docker's logs?

**Reply by parulbhagat-soprabanking** [reply]
No, logs are not getting generated. It just stops after a few seconds. Snapshot attached.
We tried running the mockoon sample file on docker and it works fine.  Sample file used: https://raw.githubusercontent.com/mockoon/mock-samples/main/samples/generate-mock-data.json
The only difference between this file and ours is that this file has request created using mockoon templating feature and ours is created without templating, just plain json file.
![72f3f8f6-ad93-4a6a-b0d1-c6ae0ce346a6](https://github.com/mockoon/mockoon/assets/131943336/f9cc4e68-6b24-47c5-8518-4ee639902f32)

**Reply by 255kb** [reply]
Without error meessage it's hard to tell where the issue come from. Could you send me your mockoon env file so I can try to reproduce the issue? You can share it here, or by mail at support@mockoon.com?

**Reply by 255kb** [reply]
Could you still check the logs with `docker logs --tail=50 aeac16227d22`?

**Reply by parulbhagat-soprabanking** [reply]
A good news - we were able to find out what the issue was . we removed the host and port from the settings and it worked absolutely fine.

Now we are trying one step ahead. This mockoon file is place on a private gitlab folder directory.  We were trying to run the file directly from gitlab on docker using the command : docker run -d -p 3000:3000 mockoon/cli:latest -d https://raw.githubusercontent.com/mockoon/mock-samples/main/samples/generate-mock-data.json -p 3000

But this is not working might be because the gitlab structure is private. We have the gitlab access token to access the file but not sure hot to use it in the above command. Could you please help us out with this issue ?

**Reply by 255kb** [reply]
Glad you found the issue. When you dockerize the mock you have to make sure that it will listen to the right host/port that is exposed by Docker. 

Concerning Gitlab authentication, it's a bit out of the scope of what Mockoon does. 
But it should be possible by using this [API endpoint](https://docs.gitlab.com/ee/api/repository_files.html#get-raw-file-from-repository) authenticated with a [personal access token](https://docs.gitlab.com/ee/api/rest/index.html#personalprojectgroup-access-tokens).

**Comment by parulbhagat-soprabanking** [comment]
We were able to run mockoon via Docker. Thanks
